### PR TITLE
test(*): fix tests involving `submit` on Chrome 60

### PR DESCRIPTION
**Pull request by gkalpak**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Test fix.

**What is the current behavior? (You can also link to an open issue here)**
Some tests involving `submit` fail on Windows with Chrome 60.

**What is the new behavior (if this is a feature change)?**
All tests pass.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] ~~Docs have been added / updated (for bug fixes / features)~~

**Other information**:
On Chrome 60 (at least on Windows) the `submit` event when clicking on a submit button is not fired on the form element, unless it is already part of the DOM.

### CODE REVIEW

File: test/ng/directive/formSpec.js [hunk 1]
```diff
@@ -420,15 +423,18 @@ describe('form', function() {
       inject(function($timeout) {
         doc = jqLite('<div>' +
                         '<form ng-submit="submitMe()">' +
-                          '<button ng-click="destroy()"></button>' +
+                          '<button type="submit" ng-click="destroy()"></button>' +
```
Comment (Narretz) [inline test/ng/directive/formSpec.js]: Is this necessary? I think if there's no other button, then the first button is the submit button

File: test/ng/directive/formSpec.js [hunk 2]
```diff
@@ -386,6 +386,9 @@ describe('form', function() {
       doc = jqLite('<form ng-submit="submitMe()">' +
                      '<input type="submit" value="submit">' +
                    '</form>');
+      // Support: Chrome 60+ (on Windows)
+      // We need to add the form to the DOM in order for `submit` events to be properly fired.
+      window.document.body.appendChild(doc[0]);
```
Comment (Narretz) [inline test/ng/directive/formSpec.js]: Aren't all of these memory leaks? Or is there some global cleanup between tests?

File: test/ng/directive/formSpec.js [hunk 3]
```diff
@@ -460,6 +466,7 @@ describe('form', function() {
         .runs(function() {
           expect(doc.html()).toBe('');
           expect(destroyed).toBe(true);
+          expect(reloadPrevented).toBe('never called');
```
Comment (Narretz) [inline test/ng/directive/formSpec.js]: Is this unrelated to the other changes?

File: test/ng/directive/formSpec.js (hunk 2 above)
Comment (gkalpak) [inline test/ng/directive/formSpec.js]: We [empty body][1] before each test.

[1]: https://github.com/angular/angular.js/blob/3650723501ca5f50003192bc4f94c77cef711e1a/test/helpers/testabilityPatch.js#L23

File: test/ng/directive/formSpec.js (hunk 1 above)
Comment (gkalpak) [inline test/ng/directive/formSpec.js]: No, it is not necessary. Just making it more explicit for people that might not know.

File: test/ng/directive/formSpec.js (hunk 3 above)
Comment (gkalpak) [inline test/ng/directive/formSpec.js]: Yes :grin:

Comment (Narretz) [inline test/ng/directive/formSpec.js]: The test results disagree with this expectation 
👀
⭕️ 

And based on the description I also would expect that this is true.

Comment (gkalpak) [inline test/ng/directive/formSpec.js]: They agree locally (probably a windows thing) 😕

Comment (gkalpak) [inline test/ng/directive/formSpec.js]: So, it turns out the cause is again that Chrome 60+ on Windows does not fire `submit` events if the form is not attached to the DOM. Changed the expectation to verify that the `submit` event is eithr never fired or (if fired) the reload is prevented.

**Review by Narretz — APPROVED** [review]
LGTM. We should obviously try to test on Chrome 60 on Travis some time soon.

**Comment by Narretz** [comment]
Looks like chrome 60 is not yet available

**Comment by gkalpak** [comment]
Yeah. Trying 59 :smiley:

**Comment by gkalpak** [comment]
Also updated to latest version for Chrome (59) and Firefox (54) available on Saucelabs.
Merging...

**Pull request closed without merging**